cc-5709: Airtime Analyzer

asantoni opened the pull request:
Putting airtime_analyzer and all the PHP changes up for review. We might only end up merging this code into 2.6.x, but it'll be easier to review it as a diff against 2.5.x for now.

naomiaro commented on python_apps/airtime_analyzer/airtime_analyzer/metadata_analyzer.py:
```diff
@@ -0,0 +1,152 @@
+        try:
+            track_number = audio_file["tracknumber"]
+            if isinstance(track_number, list): # Sometimes tracknumber is a list, ugh 
+                track_number = track_number[0]
```
wondering what else is in this list if anything?

naomiaro commented on airtime_mvc/public/js/airtime/library/plupload.js:
```diff
@@ -1,12 +1,24 @@
-		chunk_size      : '5mb',
+		url             :  baseUrl+'rest/media',
+		//chunk_size      : '5mb', //Disabling chunking since we're using the File Upload REST API now
 		unique_names    : 'true',
```
this unique names param might need to be an actual boolean?
http://www.plupload.com/docs/Options#unique_names

naomiaro commented on airtime_mvc/public/js/airtime/library/plupload.js:
```diff
@@ -1,12 +1,24 @@
+		url             :  baseUrl+'rest/media',
+		//chunk_size      : '5mb', //Disabling chunking since we're using the File Upload REST API now
 		unique_names    : 'true',
 		multiple_queues : 'true',
```
I don't see this option anymore, not sure what it used to do.

naomiaro commented on airtime_mvc/public/js/airtime/library/plupload.js:
```diff
@@ -1,12 +1,24 @@
+		UNKNOWN : { message: $.i18n._("Unknown")}
+	};
+	if (Object.freeze) {
+		Object.freeze(self.IMPORT_STATUS_CODES);
```
This wouldn't actually freeze the objects with the message status since this method is shallow.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        try {
+            //Copy the temporary file over to the "organize" folder so that it's off our webserver
+            //and accessible by airtime_analyzer which could be running on a different machine.
+            $newTempFilePath = Application_Model_StoredFile::copyFileToStor($tempFilePath, $originalFilename);
```
Is this going to have a problem if I upload
/home/naomiaro/folder1/Track1.mp3
/home/naomiaro/folder2/Track1.mp3

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+                $defaultOwner = CcSubjsQuery::create()
+                    ->filterByDbType('A')
+                    ->orderByDbId()
+                    ->findOne();
```
Maybe there's a way to create a key for each user eventually? Then they can use their own key to upload to their folder.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+            if ($storedFile->existsOnDisk()) {
+                $storedFile->delete(); //TODO: This checks your session permissions... Make it work without a session?
+            }
+            $file->delete();
```
$storedFile->delete() and $file->delete() is needed?

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        
+        $id = $this->getId();
+        if (!$id) {
+            return;
```
different response needed here?

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        $whiteList = $this->removeBlacklistedFieldsFromRequestData($requestData);
+
+        if (!$this->validateRequestData($file, $whiteList)) {
+            $file->save();
```
make sure $file is not null

naomiaro replied:
is there a point to even saving the file if nothing happens to it?

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        }
+        
+        $file = CcFilesQuery::create()->findPk($id);
+
```
maybe an error response should be returned here if $file is null, nothing to update.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+                $fullPath = $requestData["full_path"];
+                $storDir = Application_Model_MusicDir::getStorDir()->getDirectory();
+                $pos = strpos($fullPath, $storDir);
+                
```
not sure if we're allowing watched directories? Or just adding a local directory, and not watching it for now.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+                ->setHttpResponseCode(200)
+                ->appendBody(json_encode($this->sanitizeResponse($file)));
+        } else {
+            $file->setDbImportStatus(2)->save();
```
this doesn't make sense saving something here since $file won't exist.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        
+        $file = CcFilesQuery::create()->findPk($id);
+
+        $requestData = json_decode($this->getRequest()->getRawBody(), true);
```
might want to remove default zend params just in case they clash sometime.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+
+        if (!$this->validateRequestData($file, $whiteList)) {
+            $file->setDbTrackTitle($_FILES["file"]["name"]);
+            $file->setDbUtime(new DateTime("now", new DateTimeZone("UTC")));
```
should set last modified time as well, but then I guess this won't matter too much if we're merging this into 2.6.x before using it.

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+            $now  = new DateTime("now", new DateTimeZone("UTC"));
+            $file->setDbTrackTitle($_FILES["file"]["name"]);
+            $file->setDbUtime($now);
+            $file->setDbHidden(true);
```
set last modified

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        }
+
+        $id = $this->getId();
+        if (!$id) {
```
these checks seem redundant now since the getId method already returns a bad response if (!id)

naomiaro commented on airtime_mvc/application/modules/rest/controllers/MediaController.php:
```diff
@@ -0,0 +1,417 @@
+        {
+            array_push($files_array, $this->sanitizeResponse($file));
+        }
+        
```
should probably have pagination in case people have like 10,000 files..

naomiaro commented on airtime_mvc/application/controllers/PluploadController.php:
```diff
@@ -18,33 +17,62 @@ public function indexAction()
+        $limit = isset($_GET['iDisplayLength']) ? $_GET['iDisplayLength'] : 10;
+        $rowStart = isset($_GET['iDisplayStart']) ? $_GET['iDisplayStart'] : 0;
+        
+        $recentUploadsQuery = CcFilesQuery::create()->filterByDbUtime(array('min' => time() - 30 * 24 * 60 * 60))
```
is this automatically converting a number to a timestamp somewhere? kind of confusing.

naomiaro commented on airtime_mvc/application/controllers/PluploadController.php:
```diff
@@ -18,33 +17,62 @@ public function indexAction()
+        }
+        
+        $limit = isset($_GET['iDisplayLength']) ? $_GET['iDisplayLength'] : 10;
+        $rowStart = isset($_GET['iDisplayStart']) ? $_GET['iDisplayStart'] : 0;
```
should probably use the zend way to get params

naomiaro commented on airtime_mvc/application/controllers/PluploadController.php:
```diff
@@ -18,33 +17,62 @@ public function indexAction()
+        $numTotalRecentUploads = $recentUploadsQuery->find()->count();
+        
+        if ($filter == "pending") {
+            $recentUploadsQuery->filterByDbImportStatus("1");
```
shouldn't need the quotes around the 1 if it's an integer column.

naomiaro commented on airtime_mvc/application/controllers/PluploadController.php:
```diff
@@ -18,33 +17,62 @@ public function indexAction()
+        $recentUploadsQuery = CcFilesQuery::create()->filterByDbUtime(array('min' => time() - 30 * 24 * 60 * 60))
+                            ->orderByDbUtime(Criteria::DESC);
+        
+        $numTotalRecentUploads = $recentUploadsQuery->find()->count();
```
can just use count() without find.

naomiaro commented on airtime_mvc/application/controllers/PluploadController.php:
```diff
@@ -18,33 +17,62 @@ public function indexAction()
+        }
+        
+        $recentUploads = $recentUploadsQuery->offset($rowStart)->limit($limit)->find();
+        
```
should use a transaction here to make sure the count/find match.

naomiaro commented on airtime_mvc/application/controllers/PluploadController.php:
```diff
@@ -18,33 +17,62 @@ public function indexAction()
+        } else if ($filter == "failed") {
+            $recentUploadsQuery->filterByDbImportStatus(array('min' => 100));
+        }
+        
```
datatables expects another count after filtering for the iTotalDisplayRecords field.

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Fixed airtime-upgrade.php...

drigato committed:
CC-5802: Upgrade application.ini file

Execute raw sql upgrade file at the end in case errors occured during the upgrade before it gets executed

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

drigato committed:
Moved some comments around

drigato committed:
CC-5802: Upgrade application.ini file

Changed where we insert new application.ini file lines

drigato committed:
Moved deleting the maintenace temporary file flag to very end of upgrade task

drigato committed:
CC-5802: Upgrade application.ini file

drigato committed:
CC-5807: Disk Space will be updated when file is not actually deleted

Commit:
CC-5820: Airtime Analyzer: Length and MIME parsing skipped if file has
no audio metadata tags.

drigato committed:
CC-5821: Airtime Analyzer: Ftp-uploaded folder doesn't get removed from organize

$_FILES does not store the original file path so we were losing the folder name.
Fixed by explicitly passing the full filepath in via the ftp-upload-hook.sh script

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

drigato committed:
CC-5821: Airtime Analyzer: Ftp-uploaded folder doesn't get removed from organize

Was using wrong position when using substr()

drigato committed:
CC-5821: Airtime Analyzer: Ftp-uploaded folder doesn't get removed from organize

Delete the empty folder after files have been copied into imported

drigato committed:
Logging statement. To remove later.

drigato committed:
CC-5821: Airtime Analyzer: Ftp-uploaded folder doesn't get removed from organize

drigato committed:
CC-5821: Airtime Analyzer: Ftp-uploaded folder doesn't get removed from organize

Cleaner way of removing empty sub folders after ftp uploads

drigato committed:
CC-5821: Airtime Analyzer: Ftp-uploaded folder doesn't get removed from organize

Imported file path had album name twice - fixed

drigato committed:
CC-5806: Airtime Analyzer: Please implement "remove all files"

Commit:
Fixed the upstart config

Commit:
CC-5823: Airtime Analyzer: Wav file uploading fails with zero length and
empty mime

* Fixed

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
CC-5824: Airtime Analyzer: Flac file was uploaded but AA says it's
failed

* Fixed

Commit:
Fixed python-magic dependency

drigato committed:
CC-5806: Airtime Analyzer: Please implement "remove all files"

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Remove database credentials from application.ini

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Removed this silly airtime-update-db-settings script

* No longer needed, was a horrible hack in the first place.

drigato committed:
CC-5802: Upgrade application.ini file

Remove this upgrade step from the upgrade controller

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

drigato committed:
Changed GetDiskUsage so it will always return 0 if the value has not been set yet

Commit:
Fixed bitrate import in airtime_analyzer

Commit:
CC-5834: Airtime Analyzer: Bitrate is missing

* Fixed

Commit:
Some defensive code against multiprocess related deadlocks

* Reinitialize logging in child processes so we don't inherit locks.
  Might be causing a deadlock we're seeing on Pro right now.

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/locale/hu_HU/LC_MESSAGES/airtime.po
	airtime_mvc/locale/nl_NL/LC_MESSAGES/airtime.po

Commit:
Alternate bugfix for logging deadlock in airtime_analyzer

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

drigato committed:
Removing something for testing

drigato committed:
Adding back analyzer deadlock workaround.
Removing command to delete empty sub folders inside organize.

drigato committed:
CC-5840: Add Media -> Endless retries if there is a validation error

Fixed by not adding message to retry queue if request returns a
validation specific error code (422)

drigato committed:
CC-5844: Airtime Analyzer: Auto format the 'year' tag.

drigato committed:
CC-5855: Output to zendlog when files are deleted from library

Commit:
SAAS-436: Fix widgets

drigato committed:
CC-5853: Tracks marked as 'hidden' won't be marked as unavailable in Now Playing page

Commit:
SAAS-439: Genres longer than 64 characters cause Media API exception

* Truncate the genre field in the Media API

Commit:
Handle error case that could cause invalid disk usage reading

drigato committed:
CC-5859: Airtime Analyzer: format fields before writing to db

Commit:
CC-5860: Analyzer HTTP requests can hang indefinitely

* Log a backtrace when sent the USR2 signal (kill -SIGUSR2 <pid>)
* Rigged up up something to strace and restart when we see a request hanging

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
CC-5862: Invalid UTF-8 chars cause DB error

* Strip and validate UTF-8 strings in the Media API
* Also properly parse track numbers containing "-"

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Commit:
CC-5863: Deleted files still shown in Recent Uploads view

* Fixed

Commit:
CC-5861: Long file paths will cause analyzer error

* Fixed it

Commit:
CC-5856: Analyzer metadata exception

* Fixed parsing of metadata when there's empty lists returned (mutagen's
  API is awful)
* Return HTTP 422 if there's any exception in validating the metadata

Commit:
Don't retry Analyzer HTTP requests if the web app is broken

Commit:
Fixed upload file count being wrong in Add Media page (lemma to CC-5863)

Commit:
CC-5836: Status page still reports media-monitor even AA enabled

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/locale/hu_HU/LC_MESSAGES/airtime.po

Commit:
Fix error due to bad exception throwing

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/controllers/LoginController.php
	airtime_mvc/application/controllers/plugins/Acl_plugin.php
	airtime_mvc/locale/de_AT/LC_MESSAGES/airtime.po
	airtime_mvc/locale/en_GB/LC_MESSAGES/airtime.po
	airtime_mvc/locale/pt_BR/LC_MESSAGES/airtime.po

drigato committed:
SAAS-447: Cloud Storage

Added resource_id column to cc_files table

drigato committed:
Set resource_id and directory to 'cloud' type on PUT request

drigato committed:
Modified config file variables to describe not only rabbitmq settings
Removed filemover_analyzer functionality and replaced with an uploader function into Amazon s3

Commit:
Better FileMover permissions unit test (passes as root)

drigato committed:
CC-5891: Install apache-libcloud under virtualenv

drigato committed:
CC-5884: Modify Pypo -> Download files from cloud storage

drigato committed:
CC-5884: Modify Pypo -> Download files from cloud storage

drigato committed:
CC-5884: Modify Pypo -> Download files from cloud storage

drigato committed:
CC-5892: Handle when a file is downloaded from the Airtime library

drigato committed:
CC-5885: Factor out cloud storage code into separate class

drigato committed:
Prevent analyzer from crashing if it is reporting a failure to Airtime

drigato committed:
CC-5888: Handle file deletion if the file is stored in the cloud

Commit:
Use track metadata from Airtime in playout engine

* Resolves CC-5893: Metadata not updated on Airtime Pro
* Report track metadata in the schedule API, and make pypo pass that
  along to Liquidsoap via annotations.
* Move HTTP response sanitization for file metadata out of the REST module and into CcFiles
* Slightly improved the terrible exception handling in pypo

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/Bootstrap.php
	airtime_mvc/locale/de_AT/LC_MESSAGES/airtime.po
	airtime_mvc/locale/en_GB/LC_MESSAGES/airtime.po
	airtime_mvc/locale/pt_BR/LC_MESSAGES/airtime.po

drigato committed:
CC-5888: Handle file deletion if the file is stored in the cloud

drigato committed:
CC-5888: Handle file deletion if the file is stored in the cloud

drigato committed:
Small code cleanup

drigato committed:
Use a different libcloud function for uploading files

Commit:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Commit:
Merge branch 'cc-5709-airtime-analyzer-cloud-storage' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
CC-5895: Upgrade Propel to 1.7

Used composer to download Propel 1.7
The Propel library will no longer be stored in the Airtime repo

drigato committed:
CC-5895: Upgrade Propel to 1.7

Update install scripts with new propel library path

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

Using Class Table Inheritance - delegate behaviour, created a cloud_file
table to store cloud files

drigato committed:
CC-5895: Upgrade Propel to 1.7

Remove propel library from repo

drigato committed:
CC-5895: Upgrade Propel to 1.7

Moved composer.json to top level of repo so propel will get installed there
Updated install scripts to reflect this change

drigato committed:
CC-5895: Upgrade Propel to 1.7

Add vendor directory to include path

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

Fixed broken metadata display on Library page
Fixed broken download/preview option from Library page

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

For cloud files, return the original filename when downloading a file instead of the cloud object name

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

Added delete on cascade for cloud files

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

drigato committed:
CC-5896: Store cloud files in separate table, inherited from cc_files

Refactored storedfile->delete()
Added a deletePhysicalFile function to CcFile and CloudFile
Cleaned up schedule events that get passed to Pypo

drigato committed:
CC-5888: Handle file deletion if the file is stored in the cloud

Execute a python script that deletes a file from the cloud

drigato committed:
Removed media api delete-success callback because it's no longer in use

Commit:
Fix zend routing on Trusty (weird error due to REST module)

drigato committed:
Removed exiting pypo if error occurs during cloud storage downloads

drigato committed:
Audio preview fix

drigato committed:
Audio preview fix

drigato committed:
CC-5888: Handle file deletion if the file is stored in the cloud

drigato committed:
Removed cloud file deletion from Airtime Analyzer. Deletion is done from PHP now.

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/Bootstrap.php
	airtime_mvc/application/controllers/LoginController.php
	airtime_mvc/application/controllers/plugins/Acl_plugin.php

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/Bootstrap.php
	airtime_mvc/application/controllers/plugins/Acl_plugin.php

drigato committed:
SAAS-470: Zend Rest module doesn't work on Ubuntu 14.04

The Bootstrap.php in the modules directory didn't like
short php tags.

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/controllers/PluploadController.php
	airtime_mvc/application/views/scripts/plupload/index.phtml

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Commit:
Reworked upstart config for airtime_analyzer

drigato committed:
Fixed a fresh install bug that wasn't using composer installs.
Fixed a cloud file download bug where it was downloading the wrong track.
Working on amazon s3 signed urls for private objects.
Added cloud storage configuration to install script.

drigato committed:
Replaced spaces with dashes in resource_id generation
Fixed track preview and download signed urls
Removed cloud_storage_deleter.py file
Changed cloud_storage settings in airtime.conf

Commit:
CC-5929: Certain long filenames can result in the extension getting cut off

drigato committed:
Stopped returning filesize in delete functions.
Created an Amazon S3 wrapper class around Zend_Service_Amazon_S3 to add more functionality.

Commit:
Fix invalid python in StatusReporter

Commit:
Fix rare exception during shutdown in airtime_analyzer

Commit:
StatusReporter exception handler to prevent thread from dying under any circumstances

drigato committed:
Removed amazon s3 signature from pypo cache file name.
Stopped pyppo from downloading amazon s3 files if it already exists in the pypo cache dir.

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	python_apps/airtime_analyzer/airtime_analyzer/filemover_analyzer.py

drigato committed:
Reverted pypo's copy file method to how it was without cloud files
and instead directly call the cloud file's downloader method.

drigato committed:
Check filesize is greater than zero before deleteing a file from library

drigato committed:
Started commenting CloudStorageUploader class

drigato committed:
Commented cloud storage classes

Commit:
Merge pull request #75 - Saas relder csrf delete file

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

drigato committed:
Added comments to CcFiles and CloudFile classes.

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
Removed unused route from rest Media Controller

drigato committed:
Removed delete-success view script because action no longer exists.

drigato committed:
Made AirtimeAnalyzerServer.read_config_file static so CloudStorageUploader can use it.
Reverted MessageListener.msg_received_callback and MessageListener.spawn_analyzer_process back to static methods.
Moved cloud storage settings out of MessageListener and into CloudStorageUploader

drigato committed:
Changed Amazon S3 download URL format so it works with S3_EU_WEST region.
Updated amazon config file path in the downloader class

drigato committed:
Added apache-libcloud dependency to airtime_analyzer

drigato committed:
Removed composer.lock from gitignore

drigato committed:
composer.lock

drigato committed:
Change back rabbitmq config file param name

drigato committed:
rmq_config_path fix

drigato committed:
Bug fix

dpsommer committed:
Fixed 'clear' button not working when files failed to import

Commit:
Updated requirements for airtime_analyzer to make python-requests
support SNI properly

drigato committed:
Changed print statements so they go to the pypo log

drigato committed:
code cleanup

Commit:
Disable those new requirements because pip requires gcc to install them

drigato committed:
Code cleanup

drigato committed:
Fix exception thrown when the analyzer doesn't return filesize with the metadata

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	python_apps/airtime_analyzer/setup.py

Commit:
airtime_analyzer status_reporter shutdown fix

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Fix for incorrect watchdog disabling

drigato committed:
Fix bad merge

drigato committed:
Better error handling on media PUT request

Commit:
Improved some comments

drigato committed:
Merge branch 'cc-5709-airtime-analyzer-cloud-storage' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	airtime_mvc/application/modules/rest/controllers/MediaController.php

drigato committed:
SAAS-491: Edit meta data not working

Fixed: Propel 1.7 does not cast string types to integers

Commit:
CC-5950: Fix for issue where clear button in Recent Uploads didn't work
sometimes

Commit:
Slightly more robust fix for the last issue

Commit:
Actually delete files from the database via the media REST API

Commit:
Fix bad merge

Commit:
Remove clearAction from the media REST API for security

Commit:
Remove clearAction from the media REST API for security

drigato committed:
SAAS-489: Cannot preview tracks stored on disk

drigato committed:
Merge branch 'cc-5709-airtime-analyzer-cloud-storage' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/Bootstrap.php

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	airtime_mvc/application/models/StoredFile.php
	airtime_mvc/application/modules/rest/controllers/MediaController.php
	python_apps/airtime_analyzer/airtime_analyzer/analyzer_pipeline.py

Commit:
Report an error and die if amazon.conf is missing

Commit:
Error handling for if propel isn't found / composer wasn't run

drigato committed:
Close the session when a track is previewed or downloaded.
Close the file pointer when we are down with it.

drigato committed:
SAAS-504: Store provider in db

drigato committed:
Merge branch 'cc-5709-airtime-analyzer-cloud-storage' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
Merge conflict

drigato committed:
SAAS-501: Re-jig cloud_storage.conf

drigato committed:
SAAS-502: Analyzer -> Set the station id and domain in the cloud object's metadata

Set the domain name in the cloud object's metadata

drigato committed:
SAAS-505: Extract Amazon_S3 class and have it inherit from a general 'cloud backend' class

drigato committed:
SAAS-505: Extract Amazon_S3 class and have it inherit from a general 'cloud backend' class

Fixed reading credentials in from cloud_storage.conf

drigato committed:
SAAS-503: PYPO -> Use the REST API to download files

Removed Amazon S3 specific code

drigato committed:
SAAS-503: PYPO -> Use the REST API to download files

drigato committed:
SAAS-503: PYPO -> Use the REST API to download filesSAAS-503

Added python library - requests - to pybundle

drigato committed:
SAAS-510: PYPO -> Files are downloaded even if they already exist in tmp cache dir

drigato committed:
SAAS-509: PYPO can't download files via the REST API

drigato committed:
SAAS-516: Replace Zend_Service_Amazon_S3 with AWS SDK for PHP

drigato committed:
SAAS-515: Prefix object names with station name

Commit:
Silan cue-in/out analysis now moved to airtime_analyzer

* Added a new CuePointAnalyzer using silan (will remove silan stuff from pypo later)
* Makes silan analysis slightly more reliable for certain short files.
* Fixes CC-5961: Audio duration cutoff for WAVE files
* Added unit tests for the new analyzer and improved code coverage slightly
* Added unit tests for WAVE metadata extraction and an invalid WMA file

Commit:
ReplayGain analysis in airtime_analyzer using python-rgain

* Implemented Replaygain analysis for Ogg Vorbis, MP3, MP4, and FLAC
  using python-rgain
* Added unit tests for ReplayGainAnalyzer
* Squashed stderr output in ReplayGainAnalyzer and CuePointAnalyzer
* Clean up

Commit:
Removed all ReplayGain and Silan stuff from pypo, plus a bugfix

* Ripped out all the ReplayGain and Silan analysis from pypo, since
  it's now implemented in airtime_analyzer. This fixes a bunch of
  race conditions.
* Also renamed the replaygain field to replay_gain in airtime_analyzer
  to match Airtime.

Commit:
CC-5739: Added Liquidsoap playability test to airtime_analyzer

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	python_apps/airtime_analyzer/airtime_analyzer/analyzer_pipeline.py
	python_apps/airtime_analyzer/setup.py

drigato committed:
Fixed merge error

drigato committed:
SAAS-514: PYPO -> Tracks may not become ready in time for playout

drigato committed:
SAAS-519: webstreams not being deleted

Commit:
Close file descriptors in airtime_analyzer before forking subprocesses

drigato committed:
Updated propel models

Commit:
Add a comment about the proxy pattern

drigato committed:
SAAS-527: Allow files to be uploaded to either the cloud or on local file storage

Not quite done.

drigato committed:
Merge branch 'cc-5709-airtime-analyzer-cloud-storage' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
SAAS-527: Allow files to be uploaded to either the cloud or on local file storage

Done.

Commit:
Cloud storage cleanup and fixed the unit tests

Commit:
Stub CloudStorageAnalyzer tests

Commit:
Unit test tweak for CloudStorage

Commit:
Added missing config_File.py

drigato committed:
Small analyzer fix

drigato committed:
SAAS-529: Replace Apache Libcloud with Python-Boto SDK

drigato committed:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
SAAS-536: Check for future scheduled files uses cloud_file id instead of cc_files id

drigato committed:
SAAS-539: Analyzer can return values in scientific notation which are not compatible with Airtime

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
SAAS-539: Analyzer can return values in scientific notation which are not compatible with Airtime

Fix unit tests

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
SAAS-542: ryerson.airtime.pro file doesn't play, preview, or download

drigato committed:
SAAS-542: ryerson.airtime.pro file doesn't play, preview, or download

dpsommer committed:
Initial commit for update to ACL for REST module; NEEDS TESTING

dpsommer committed:
Moved REST authorization boilerplate to Zend Acl_plugin

drigato committed:
SAAS-555: Analyzer calculates wrong cue out

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	airtime_mvc/application/modules/rest/controllers/MediaController.php

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Conflicts:
	CREDITS

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Fixed CSRF token checking in MediaController (broken by last 2 commits)

* Also fixed getOwnerId(), no longer calls a non-existent function

Commit:
Pin python-daemon to 1.6 because 2.0.3 breaks setuptools

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Conflicts:
	airtime_mvc/build/sql/schema.sql

drigato committed:
Disabling cue point analyzer

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

drigato committed:
Enabling cuepoint analyzer because the silan bug has been fixed

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Commit:
Renabling CuePointAnalyzer

Commit:
Fixed quotation issue with playability analyzer

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-cloud-storage

Commit:
Make the unit tests pass again

Commit:
Fixed the unit tests again

Commit:
REST permission fix

Commit:
Backport ACL REST permission fix

dpsommer committed:
SAAS-595 - Changed Zend validation and added sanitization in file import process to throw out bad track number metadata

dpsommer committed:
SAAS-595 - Updated validation and sanitization

drigato committed:
SAAS-596: Store file size and hash in database

Added file size and hash to the metadata dict in the analyzer

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Simplify the metadata sanitization and bugfix it

* SAAS-376 and CC-5868

Commit:
CC-5868: Make the BPM field less strict

drigato committed:
SAAS-596: Store file size and hash in database

Removed try/except while getting file size and hash

Commit:
Pull the logic for Media REST API out of the controller

drigato committed:
SAAS-596: Store file size and hash in database

Renamed metadata md5 value to match cc_files column

Commit:
Merge branch 'cc-5709-airtime-analyzer-cloud-storage' into cc-5709-airtime-analyzer

Conflicts:
	airtime_mvc/application/models/airtime/CcFiles.php
	airtime_mvc/application/modules/rest/controllers/MediaController.php

dpsommer committed:
Fixed CSRF prevention checks for REST calls, moved CSRF initialization to Bootstrap

dpsommer committed:
Small fix to BPM sanitization

Commit:
Merge branch 'cc-5709-airtime-analyzer' of github.com:sourcefabric/Airtime into cc-5709-airtime-analyzer

Commit:
Use more secure random number generation for CSRF auth tokens

* Also cleaned up pull request

Commit:
Merge branch 'cc-5709-airtime-analyzer-csrf' into cc-5709-airtime-analyzer

Commit:
Merge branch '2.5.x' into cc-5709-airtime-analyzer

Commit:
SAAS-602: Fix getaddrinfo deadlock (again)

drigato committed:
Merge branch 'cc-5709-airtime-analyzer' into cc-5709-airtime-analyzer-store-file-size-and-hash-in-db
